### PR TITLE
[Feature] PyTorch Serialization 

### PR DESCRIPTION
**Pull request by teexone**
Added new ways to serialize tensors with Malevich Square. Using Base122 encoding and saving four things: shapes, gradients, devices and tensor data. Tested with:

```python
import torch

x = [
    torch.rand(10, 10),
    torch.rand(1, ).to('cuda')
]

from malevich.square import from_df, to_df

df = to_df(x)
y = from_df(df)

assert torch.allclose(x[0], y[0])
assert torch.allclose(x[1], y[1])

x_grad = torch.rand_like(x[0])
y_grad = torch.rand_like(y[0])
x[0].requires_grad_()
x[0].backward(torch.ones_like(x[0]))
x[0] = x[0].cpu()


df = to_df(x)
y = from_df(df)

print(y[0].grad)
assert torch.allclose(x[0], y[0])
assert torch.allclose(x[1], y[1])
assert torch.allclose(x[0].grad, y[0].grad)

assert x[0].device == y[0].device and y[0].device == torch.device('cpu'), f'{x[0].device} != {y[0].device} != cpu'
assert x[1].device == y[1].device
```

### CODE REVIEW

**Comment by teexone** [comment]
Closed as it was manually merged into dev/unstable.
